app/src/main/java/com/andrew/tasky/core/di/AppModule.kt

```diff
@@ -28,15 +33,29 @@ object AppModule {
         return EmailPatternValidatorImpl()
     }
 
-    private val okHttp = OkHttpClient.Builder().addInterceptor(ApiKeyInterceptor)
-
     @Provides
     @Singleton
     fun provideAuthApi(): AuthApi {
         return Retrofit.Builder()
-            .baseUrl("https://tasky.pl-coding.com/")
+            .baseUrl("https://tasky.pl-coding.com")
+            .client(OkHttpClient.Builder().addInterceptor(ApiKeyInterceptor).build())
+            .addConverterFactory(MoshiConverterFactory.create())
+            .build()
+            .create()
+    }
+
+    @Provides
+    @Singleton
+    fun provideAgendaApi(prefs: SharedPreferences): AgendaApi {
+        return Retrofit.Builder()
+            .baseUrl("https://tasky.pl-coding.com")
+            .client(
+                OkHttpClient.Builder()
```

Comment:
I'd create a separate provides function for this client since you need it for all your API interfaces. If you want to configure it, you would need to change it at all spots instead of just one